### PR TITLE
Read pipe cleanup refactor

### PR DESCRIPTION
**Pull request by Aryex**
### Summary

* Extracted read pipe cleanup logic into a separate class. It is intended to be re-used by JSON reader. 
* Added new supporting interfaces. 
* Simplified existing partition interfaces:
  * Previously, ParquetFileRange and VerticaDistributedFileSystem define fields which are not needed to be optional. They are refactored to normal fields. Related files are updated.

### Related Issue

#416 

### Additional Reviewers
@alexey-temnikov 
@alexr-bq 
@jeremyp-bq 
@jonathanl-bq

### CODE REVIEW

**Review by jonathanl-bq — COMMENTED** [review]
I don't have time to do a really thorough review, but I think we should avoid exposing index math in our traits. It feels like a lower level implementation detail more than something we should expose in a trait, but I don't know if there's a better justification for this current design.

**Comment by Aryex** [comment]
> I don't have time to do a really thorough review, but I think we should avoid exposing index math in our traits. It feels like a lower level implementation detail more than something we should expose in a trait, but I don't know if there's a better justification for this current design.

Updated. Also changed the names of the traits to Cleanup and PartitionCleanup

File: connector/src/main/scala/com/vertica/spark/datasource/partitions/PartitionCleanup.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.vertica.spark.datasource.partitions
+
+import org.apache.spark.sql.connector.read.InputPartition
+
+trait PartitionCleanup extends InputPartition {
+  def getCleanUps: Seq[Cleanup]
```
Comment (jeremyp-bq) [inline connector/src/main/scala/com/vertica/spark/datasource/partitions/PartitionCleanup.scala:19]: Minor nit: getCleanUps when we use Cleanup (up not capitalized) everywhere else.

Comment (Aryex) [reply connector/src/main/scala/com/vertica/spark/datasource/partitions/PartitionCleanup.scala:19]: Updated

**Comment by Aryex** [comment]
Updated trait name to better reflect the implementor capabilities
Cleanup -> PortionId: Implementors can identify themselves amongst other portions from the same file.
PartitionCleanup -> Cleanup: Implementors contains cleanup information.

**Review by jeremyp-bq — APPROVED** [review]

**Pull request merged**